src/dlsproto/client/request/internal/GetRange.d

```diff
@@ -398,11 +398,35 @@ private scope class GetRangeHandler
 
     private /* scope */ class RecordStream
     {
-        /// The acquired buffer to store a batch of records.
-        private void[]* batch_buffer;
+        /// Double buffer structure
+        private struct DoubleBuffer
+        {
+            /// Acquired buffers to store batches of records. Reader always
+            /// appends to the back one and the RecordStream always uses the
+            /// front one.
+            private void[]* front_buffer;
```

Comment:
You could even name them `reader_buffer` and `record_stream_buffer`, to make it super clear which is which.

Reply:
Right, or just `output` and `input`. (The `_buffer` suffix isn't needed.)